File: backend/app/schemas/cost_sheet.py
from pydantic import BaseModel
from typing import List, Optional

class PaymentMilestone(BaseModel):
    milestone_name: str
    percentage: float
    amount: float
    due_condition: str

class CostSheetCalculateRequest(BaseModel):
    lead_id: Optional[int] = None
    project_id: int
    unit_number: str
    configuration: str
    super_builtup_area_sqft: float
    base_rate_per_sqft: float
    floor_number: int = 1
    floor_rise_rate_per_sqft: float = 25.0
    plc_rate_per_sqft: float = 0.0
    car_parking_slots: int = 1
    car_parking_rate: float = 400000.0
    clubhouse_charges: float = 350000.0
    possession_charges: float = 150000.0
    gst_rate_pct: float = 5.0
    stamp_duty_rate_pct: float = 7.0
    registration_fee: float = 30000.0
    discount_amount: float = 0.0

class CostSheetBreakdown(BaseModel):
    base_selling_price: float
    floor_rise_charges: float
    plc_charges: float
    car_parking_charges: float
    discount_amount: float
    total_agreement_value: float

    clubhouse_charges: float
    possession_charges: float
    total_additional_charges: float

    gst_amount: float
    stamp_duty_amount: float
    registration_fee: float
    total_statutory_charges: float

    grand_total: float
    token_booking_amount: float # 10%

class CostSheetResponse(BaseModel):
    project_id: int
    project_name: str
    builder_name: str
    location: str
    unit_number: str
    configuration: str
    super_builtup_area_sqft: float
    base_rate_per_sqft: float
    breakdown: CostSheetBreakdown
    payment_schedule: List[PaymentMilestone]
    whatsapp_summary: str

class QuickBookFromCostSheetRequest(CostSheetCalculateRequest):
    lead_id: int
    broker_id: Optional[int] = None
    assigned_executive_id: Optional[int] = None
    token_amount_paid: Optional[float] = None
    notes: Optional[str] = None

File: frontend/src/types/costSheet.ts
export interface PaymentMilestone {
  milestone_name: string;
  percentage: number;
  amount: number;
  due_condition: string;
}

export interface CostSheetBreakdown {
  base_selling_price: number;
  floor_rise_charges: number;
  plc_charges: number;
  car_parking_charges: number;
  discount_amount: number;
  total_agreement_value: number;

  clubhouse_charges: number;
  possession_charges: number;
  total_additional_charges: number;

  gst_amount: number;
  stamp_duty_amount: number;
  registration_fee: number;
  total_statutory_charges: number;

  grand_total: number;
  token_booking_amount: number;
}

export interface CostSheetResponse {
  project_id: number;
  project_name: string;
  builder_name: string;
  location: string;
  unit_number: string;
  configuration: string;
  super_builtup_area_sqft: number;
  base_rate_per_sqft: number;
  breakdown: CostSheetBreakdown;
  payment_schedule: PaymentMilestone[];
  whatsapp_summary: string;
}

export interface CostSheetCalculateRequest {
  lead_id?: number | null;
  project_id: number;
  unit_number: string;
  configuration: string;
  super_builtup_area_sqft: number;
  base_rate_per_sqft: number;
  floor_number?: number;
  floor_rise_rate_per_sqft?: number;
  plc_rate_per_sqft?: number;
  car_parking_slots?: number;
  car_parking_rate?: number;
  clubhouse_charges?: number;
  possession_charges?: number;
  gst_rate_pct?: number;
  stamp_duty_rate_pct?: number;
  registration_fee?: number;
  discount_amount?: number;
}

export interface QuickBookFromCostSheetRequest extends CostSheetCalculateRequest {
  lead_id: number;
  broker_id?: number | null;
  assigned_executive_id?: number | null;
  token_amount_paid?: number | null;
  notes?: string | null;
}
